libusb/core.c

```diff
@@ -2204,6 +2207,108 @@ int API_EXPORTED libusb_set_auto_detach_kernel_driver(
 	return LIBUSB_SUCCESS;
 }
 
+/** \ingroup libusb_dev
+ * Check if the endpoint supports RAW_IO.
+ * \param dev_handle a device handle
+ * \param endpoint the endpoint to check
+ *
+ * \returns 1 if the endpoint supports RAW_IO
+ * \returns 0 if the endpoint does not support RAW_IO
+ * \returns a LIBUSB_ERROR code on failure
+ *
+ * Only endpoints using the WinUSB driver support RAW_IO,
+ * for all other backends/drivers this function will return 0.
+ *
+ * Fails if the interface the endpoint belongs to isn't claimed,
+ * \see libusb_claim_interface().
+ *
+ * \see libusb_endpoint_set_raw_io()
+ *
+ * Since version 1.0.28, \ref LIBUSB_API_VERSION >= 0x0100010B
+ */
+int API_EXPORTED libusb_endpoint_supports_raw_io(libusb_device_handle* dev_handle,
+	uint8_t endpoint)
+{
+	if (usbi_backend.endpoint_supports_raw_io == NULL ||
+		usbi_backend.endpoint_set_raw_io == NULL)
```

Comment:
Why the check for usbi_backend.endpoint_set_raw_io?

Reply:
Presumably because calling `libusb_endpoint_supports_raw_io` is supposed to guarantee that it's OK to call `libusb_endpoint_set_raw_io`. If the function pointer for the latter isn't set, then the RAW_IO support isn't complete.

I think the cross-check makes sense, and it should probably also check that `usbi_backend.get_max_raw_io_transfer_size` is set too.

Reply:
Agreed, shows the intent much better.